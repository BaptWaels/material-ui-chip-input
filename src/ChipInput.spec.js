/* eslint-env jest */
import React from 'react'
import { mount } from 'enzyme'
import ChipInput from './ChipInput'

describe('uncontrolled mode', () => {
  it('matches the snapshot', () => {
    const tree = mount(
      <ChipInput defaultValue={['foo', 'bar']} />
    )
    expect(tree).toMatchSnapshot()
  })

  it('displays the default value in chips', () => {
    const tree = mount(
      <ChipInput defaultValue={['foo', 'bar', 'foobar']} />
    )
    expect(tree.find('Chip').map((chip) => chip.text())).toEqual(['foo', 'bar', 'foobar'])
  })

  it('displays added chips', () => {
    const tree = mount(
      <ChipInput defaultValue={['foo', 'bar']} />
    )
    tree.find('input').getDOMNode().value = 'test'
    tree.find('input').simulate('keyDown', { keyCode: 13 }) // press enter
    expect(tree.find('Chip').map((chip) => chip.text())).toEqual(['foo', 'bar', 'test'])
  })

  it('calls onChange when adding new chips', () => {
    const handleChange = jest.fn()
    const tree = mount(
      <ChipInput onChange={handleChange} />
    )

    tree.find('input').getDOMNode().value = 'foo'
    tree.find('input').simulate('keyDown', { keyCode: 13 }) // press enter
    expect(handleChange).toBeCalledWith(['foo'])

    tree.find('input').getDOMNode().value = 'bar'
    tree.find('input').simulate('keyDown', { keyCode: 13 }) // press enter
    expect(handleChange).toBeCalledWith(['foo', 'bar'])
  })

  it('calls onChange when deleting chips with backspace key', () => {
    const handleChange = jest.fn()
    const tree = mount(
      <ChipInput defaultValue={['foo', 'bar']} onChange={handleChange} />
    )

    tree.find('input').simulate('keyDown', { keyCode: 8 }) // backspace (to focus the chip)
    tree.find('input').simulate('keyDown', { keyCode: 8 }) // backspace (to remove the chip)
    expect(handleChange).toBeCalledWith(['foo'])
  })

  it('calls onChange when deleting chips with delete key', () => {
    const handleChange = jest.fn()
    const tree = mount(
      <ChipInput defaultValue={['foo', 'bar']} onChange={handleChange} />
    )

    tree.find('input').simulate('keyDown', { keyCode: 8 }) // backspace (to focus the chip)
    tree.find('input').simulate('keyDown', { keyCode: 46 }) // del (to remove the chip)
    expect(handleChange).toBeCalledWith(['foo'])
  })

  it('calls onChange when deleting chips by clicking on the remove button', () => {
    const handleChange = jest.fn()
    const tree = mount(
      <ChipInput defaultValue={['foo', 'bar']} onChange={handleChange} />
    )
    tree.find('Cancel').first().simulate('click')
    expect(handleChange).toBeCalledWith(['bar'])
  })
})

describe('chip focusing', () => {
  function getFocusedChip (tree) {
    return tree.find('Chip').filterWhere((chip) => chip.getDOMNode().style.backgroundColor !== '')
  }

  function focusChip (tree, name) {
    tree.find('Chip').filterWhere((chip) => chip.text() === name).simulate('click')
  }

  it('focuses a chip on click', () => {
    const tree = mount(
      <ChipInput defaultValue={['foo', 'bar']} />
    )
    tree.find('Chip').at(1).simulate('click')
    expect(getFocusedChip(tree).length).toBe(1)
    expect(getFocusedChip(tree).text()).toBe('bar')
  })

  it('focuses the last chip when pressing backspace', () => {
    const tree = mount(
      <ChipInput defaultValue={['a', 'b', 'c']} />
    )
    tree.find('input').simulate('keyDown', { keyCode: 8 }) // backspace
    expect(getFocusedChip(tree).text()).toBe('c')
  })

  it('unfocuses the focused chip while adding a new chip', () => {
    const tree = mount(
      <ChipInput defaultValue={['foo', 'bar']} />
    )
    focusChip(tree, 'foo')
    tree.find('input').simulate('keyDown')
    expect(getFocusedChip(tree).length).toBe(0)
  })

  it('unfocuses the focused chip on blur', () => {
    const tree = mount(
      <ChipInput defaultValue={['foo', 'bar']} />
    )
    focusChip(tree, 'foo')
    tree.find('input').simulate('blur')
    expect(getFocusedChip(tree).length).toBe(0)
  })

  it('unfocuses the focused chip when switching to disabled state', () => {
    const tree = mount(
      <ChipInput defaultValue={['foo', 'bar']} />
    )
    focusChip(tree, 'foo')
    tree.setProps({ disabled: true })
    expect(getFocusedChip(tree).length).toBe(0)
  })

  it('moves the focus to the left when pressing the left arrow key', () => {
    const tree = mount(
      <ChipInput defaultValue={['a', 'b', 'c']} />
    )
    focusChip(tree, 'b')
    tree.find('input').simulate('keyDown', { keyCode: 37 }) // arrow left
    expect(getFocusedChip(tree).text()).toBe('a')

    // keep the first element focused when pressing left if the first element is already focused
    tree.find('input').simulate('keyDown', { keyCode: 37 }) // arrow left
    expect(getFocusedChip(tree).text()).toBe('a')
  })

  it('moves the focus to the right when pressing the right arrow key', () => {
    const tree = mount(
      <ChipInput defaultValue={['a', 'b', 'c']} />
    )
    focusChip(tree, 'b')
    tree.find('input').simulate('keyDown', { keyCode: 39 }) // arrow right
    expect(getFocusedChip(tree).text()).toBe('c')

    // onfocus all chips if the right arrow key is pressed when focusing the last chip
    tree.find('input').simulate('keyDown', { keyCode: 39 }) // arrow right
    expect(getFocusedChip(tree).length).toBe(0)
  })

  it('focuses the chip to the left when removing a chip by pressing backspace', () => {
    const tree = mount(
      <ChipInput defaultValue={['a', 'b', 'c']} />
    )
    focusChip(tree, 'b')

    tree.find('input').simulate('keyDown', { keyCode: 8 }) // backspace
    expect(getFocusedChip(tree).text()).toBe('a')
  })

  it('focuses the chip at the previously focused position when removing a chip by pressing delete', () => {
    const tree = mount(
      <ChipInput defaultValue={['a', 'b', 'c']} />
    )
    focusChip(tree, 'b')

    tree.find('input').simulate('keyDown', { keyCode: 46 }) // delete
    expect(getFocusedChip(tree).text()).toBe('c')
  })
})

describe('placeholder', () => {
  it('displays a placeholder', () => {
    const tree = mount(
      <ChipInput placeholder='Placeholder' />
    )

    expect(tree.find('input').getDOMNode().getAttribute('placeholder')).toBe('Placeholder')
  })

  it('is hidden if there are chips', () => {
    const tree = mount(
      <ChipInput placeholder='Placeholder' value={['foo']} />
    )

    expect(tree.find('input').getDOMNode().getAttribute('placeholder')).toBe(null)
  })

  it('is hidden if there is a floating label', () => {
    const tree = mount(
      <ChipInput placeholder='Placeholder' label='Floating label' />
    )

    expect(tree.find('input').getDOMNode().getAttribute('placeholder')).toBe(null)
  })

  it('is visible if the floating label is floating', () => {
    const tree = mount(
      <ChipInput placeholder='Placeholder' label='Floating label' />
    )
    tree.find('input').simulate('focus')
    expect(tree.find('InputLabel').prop('shrink')).toBe(true)
    expect(tree.find('input').getDOMNode().getAttribute('placeholder')).toBe('Placeholder')
  })
})

describe('floating label', () => {
  it('is displayed', () => {
    const tree = mount(
      <ChipInput label='Floating label' />
    )
    expect(tree.find('InputLabel').text()).toBe('Floating label')
    expect(tree.find('InputLabel').prop('shrink')).toBe(false)
  })

  it('shrinks if there are chips', () => {
    const tree = mount(
      <ChipInput label='Floating label' value={['foo']} />
    )
    expect(tree.find('InputLabel').prop('shrink')).toBe(true)
  })

  it('shrinks if there are no chips but text input', () => {
    const tree = mount(
      <ChipInput label='Floating label' />
    )
    tree.find('input').getDOMNode().value = 'foo'
    tree.find('input').simulate('change')
    expect(tree.find('InputLabel').prop('shrink')).toBe(true)
  })
})
