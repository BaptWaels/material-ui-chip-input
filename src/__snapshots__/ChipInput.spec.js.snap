// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uncontrolled mode matches the snapshot 1`] = `
<WithStyles(ChipInput)
  defaultValue={
    Array [
      "foo",
      "bar",
    ]
  }
>
  <ChipInput
    allowDuplicates={false}
    blurBehavior="clear"
    classes={
      Object {
        "chip": "ChipInput-chip-14",
        "chipContainer": "ChipInput-chipContainer-4",
        "disabled": "ChipInput-disabled-11",
        "error": "ChipInput-error-13",
        "focused": "ChipInput-focused-10",
        "helperText": "ChipInput-helperText-8",
        "inkbar": "ChipInput-inkbar-9",
        "input": "ChipInput-input-3",
        "inputRoot": "ChipInput-inputRoot-2",
        "label": "ChipInput-label-6",
        "labelShrink": "ChipInput-labelShrink-7",
        "labeled": "ChipInput-labeled-5",
        "root": "ChipInput-root-1",
        "underline": "ChipInput-underline-12",
      }
    }
    clearInputValueOnChange={false}
    defaultValue={
      Array [
        "foo",
        "bar",
      ]
    }
    newChipKeyCodes={
      Array [
        13,
      ]
    }
  >
    <WithStyles(FormControl)
      className="ChipInput-root-1"
      onClick={[Function]}
    >
      <FormControl
        className="ChipInput-root-1"
        classes={
          Object {
            "fullWidth": "MuiFormControl-fullWidth-18",
            "marginDense": "MuiFormControl-marginDense-17",
            "marginNormal": "MuiFormControl-marginNormal-16",
            "root": "MuiFormControl-root-15",
          }
        }
        component="div"
        disabled={false}
        error={false}
        fullWidth={false}
        margin="none"
        onClick={[Function]}
        required={false}
        variant="standard"
      >
        <div
          className="MuiFormControl-root-15 ChipInput-root-1"
          onClick={[Function]}
        >
          <div
            className="ChipInput-chipContainer-4 ChipInput-inkbar-9 ChipInput-underline-12"
          >
            <WithStyles(Chip)
              className="ChipInput-chip-14"
              key="0"
              label="foo"
              onClick={[Function]}
              onDelete={[Function]}
              style={
                Object {
                  "backgroundColor": undefined,
                  "pointerEvents": undefined,
                }
              }
            >
              <Chip
                className="ChipInput-chip-14"
                classes={
                  Object {
                    "avatar": "MuiChip-avatar-31",
                    "avatarChildren": "MuiChip-avatarChildren-34",
                    "avatarColorPrimary": "MuiChip-avatarColorPrimary-32",
                    "avatarColorSecondary": "MuiChip-avatarColorSecondary-33",
                    "clickable": "MuiChip-clickable-22",
                    "clickableColorPrimary": "MuiChip-clickableColorPrimary-23",
                    "clickableColorSecondary": "MuiChip-clickableColorSecondary-24",
                    "colorPrimary": "MuiChip-colorPrimary-20",
                    "colorSecondary": "MuiChip-colorSecondary-21",
                    "deletable": "MuiChip-deletable-25",
                    "deletableColorPrimary": "MuiChip-deletableColorPrimary-26",
                    "deletableColorSecondary": "MuiChip-deletableColorSecondary-27",
                    "deleteIcon": "MuiChip-deleteIcon-39",
                    "deleteIconColorPrimary": "MuiChip-deleteIconColorPrimary-40",
                    "deleteIconColorSecondary": "MuiChip-deleteIconColorSecondary-41",
                    "deleteIconOutlinedColorPrimary": "MuiChip-deleteIconOutlinedColorPrimary-42",
                    "deleteIconOutlinedColorSecondary": "MuiChip-deleteIconOutlinedColorSecondary-43",
                    "icon": "MuiChip-icon-35",
                    "iconColorPrimary": "MuiChip-iconColorPrimary-36",
                    "iconColorSecondary": "MuiChip-iconColorSecondary-37",
                    "label": "MuiChip-label-38",
                    "outlined": "MuiChip-outlined-28",
                    "outlinedPrimary": "MuiChip-outlinedPrimary-29",
                    "outlinedSecondary": "MuiChip-outlinedSecondary-30",
                    "root": "MuiChip-root-19",
                  }
                }
                clickable={false}
                color="default"
                component="div"
                label="foo"
                onClick={[Function]}
                onDelete={[Function]}
                style={
                  Object {
                    "backgroundColor": undefined,
                    "pointerEvents": undefined,
                  }
                }
                variant="default"
              >
                <div
                  className="MuiChip-root-19 MuiChip-clickable-22 MuiChip-deletable-25 ChipInput-chip-14"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  role="button"
                  style={
                    Object {
                      "backgroundColor": undefined,
                      "pointerEvents": undefined,
                    }
                  }
                  tabIndex={0}
                >
                  <span
                    className="MuiChip-label-38"
                  >
                    foo
                  </span>
                  <pure(Cancel)
                    className="MuiChip-deleteIcon-39"
                    onClick={[Function]}
                  >
                    <Cancel
                      className="MuiChip-deleteIcon-39"
                      onClick={[Function]}
                    >
                      <WithStyles(SvgIcon)
                        className="MuiChip-deleteIcon-39"
                        onClick={[Function]}
                      >
                        <SvgIcon
                          className="MuiChip-deleteIcon-39"
                          classes={
                            Object {
                              "colorAction": "MuiSvgIcon-colorAction-47",
                              "colorDisabled": "MuiSvgIcon-colorDisabled-49",
                              "colorError": "MuiSvgIcon-colorError-48",
                              "colorPrimary": "MuiSvgIcon-colorPrimary-45",
                              "colorSecondary": "MuiSvgIcon-colorSecondary-46",
                              "fontSizeInherit": "MuiSvgIcon-fontSizeInherit-50",
                              "fontSizeLarge": "MuiSvgIcon-fontSizeLarge-52",
                              "fontSizeSmall": "MuiSvgIcon-fontSizeSmall-51",
                              "root": "MuiSvgIcon-root-44",
                            }
                          }
                          color="inherit"
                          component="svg"
                          fontSize="default"
                          onClick={[Function]}
                          viewBox="0 0 24 24"
                        >
                          <svg
                            aria-hidden="true"
                            className="MuiSvgIcon-root-44 MuiChip-deleteIcon-39"
                            focusable="false"
                            onClick={[Function]}
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                            />
                          </svg>
                        </SvgIcon>
                      </WithStyles(SvgIcon)>
                    </Cancel>
                  </pure(Cancel)>
                </div>
              </Chip>
            </WithStyles(Chip)>
            <WithStyles(Chip)
              className="ChipInput-chip-14"
              key="1"
              label="bar"
              onClick={[Function]}
              onDelete={[Function]}
              style={
                Object {
                  "backgroundColor": undefined,
                  "pointerEvents": undefined,
                }
              }
            >
              <Chip
                className="ChipInput-chip-14"
                classes={
                  Object {
                    "avatar": "MuiChip-avatar-31",
                    "avatarChildren": "MuiChip-avatarChildren-34",
                    "avatarColorPrimary": "MuiChip-avatarColorPrimary-32",
                    "avatarColorSecondary": "MuiChip-avatarColorSecondary-33",
                    "clickable": "MuiChip-clickable-22",
                    "clickableColorPrimary": "MuiChip-clickableColorPrimary-23",
                    "clickableColorSecondary": "MuiChip-clickableColorSecondary-24",
                    "colorPrimary": "MuiChip-colorPrimary-20",
                    "colorSecondary": "MuiChip-colorSecondary-21",
                    "deletable": "MuiChip-deletable-25",
                    "deletableColorPrimary": "MuiChip-deletableColorPrimary-26",
                    "deletableColorSecondary": "MuiChip-deletableColorSecondary-27",
                    "deleteIcon": "MuiChip-deleteIcon-39",
                    "deleteIconColorPrimary": "MuiChip-deleteIconColorPrimary-40",
                    "deleteIconColorSecondary": "MuiChip-deleteIconColorSecondary-41",
                    "deleteIconOutlinedColorPrimary": "MuiChip-deleteIconOutlinedColorPrimary-42",
                    "deleteIconOutlinedColorSecondary": "MuiChip-deleteIconOutlinedColorSecondary-43",
                    "icon": "MuiChip-icon-35",
                    "iconColorPrimary": "MuiChip-iconColorPrimary-36",
                    "iconColorSecondary": "MuiChip-iconColorSecondary-37",
                    "label": "MuiChip-label-38",
                    "outlined": "MuiChip-outlined-28",
                    "outlinedPrimary": "MuiChip-outlinedPrimary-29",
                    "outlinedSecondary": "MuiChip-outlinedSecondary-30",
                    "root": "MuiChip-root-19",
                  }
                }
                clickable={false}
                color="default"
                component="div"
                label="bar"
                onClick={[Function]}
                onDelete={[Function]}
                style={
                  Object {
                    "backgroundColor": undefined,
                    "pointerEvents": undefined,
                  }
                }
                variant="default"
              >
                <div
                  className="MuiChip-root-19 MuiChip-clickable-22 MuiChip-deletable-25 ChipInput-chip-14"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  role="button"
                  style={
                    Object {
                      "backgroundColor": undefined,
                      "pointerEvents": undefined,
                    }
                  }
                  tabIndex={0}
                >
                  <span
                    className="MuiChip-label-38"
                  >
                    bar
                  </span>
                  <pure(Cancel)
                    className="MuiChip-deleteIcon-39"
                    onClick={[Function]}
                  >
                    <Cancel
                      className="MuiChip-deleteIcon-39"
                      onClick={[Function]}
                    >
                      <WithStyles(SvgIcon)
                        className="MuiChip-deleteIcon-39"
                        onClick={[Function]}
                      >
                        <SvgIcon
                          className="MuiChip-deleteIcon-39"
                          classes={
                            Object {
                              "colorAction": "MuiSvgIcon-colorAction-47",
                              "colorDisabled": "MuiSvgIcon-colorDisabled-49",
                              "colorError": "MuiSvgIcon-colorError-48",
                              "colorPrimary": "MuiSvgIcon-colorPrimary-45",
                              "colorSecondary": "MuiSvgIcon-colorSecondary-46",
                              "fontSizeInherit": "MuiSvgIcon-fontSizeInherit-50",
                              "fontSizeLarge": "MuiSvgIcon-fontSizeLarge-52",
                              "fontSizeSmall": "MuiSvgIcon-fontSizeSmall-51",
                              "root": "MuiSvgIcon-root-44",
                            }
                          }
                          color="inherit"
                          component="svg"
                          fontSize="default"
                          onClick={[Function]}
                          viewBox="0 0 24 24"
                        >
                          <svg
                            aria-hidden="true"
                            className="MuiSvgIcon-root-44 MuiChip-deleteIcon-39"
                            focusable="false"
                            onClick={[Function]}
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                            />
                          </svg>
                        </SvgIcon>
                      </WithStyles(SvgIcon)>
                    </Cancel>
                  </pure(Cancel)>
                </div>
              </Chip>
            </WithStyles(Chip)>
            <WithStyles(Input)
              classes={
                Object {
                  "input": "ChipInput-input-3",
                  "root": "ChipInput-inputRoot-2",
                }
              }
              disableUnderline={true}
              inputRef={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              placeholder={null}
              value=""
            >
              <Input
                classes={
                  Object {
                    "disabled": "MuiInput-disabled-56",
                    "error": "MuiInput-error-58",
                    "focused": "MuiInput-focused-55",
                    "formControl": "MuiInput-formControl-54",
                    "fullWidth": "MuiInput-fullWidth-60",
                    "input": "MuiInput-input-61 ChipInput-input-3",
                    "inputMarginDense": "MuiInput-inputMarginDense-62",
                    "inputMultiline": "MuiInput-inputMultiline-63",
                    "inputType": "MuiInput-inputType-64",
                    "inputTypeSearch": "MuiInput-inputTypeSearch-65",
                    "multiline": "MuiInput-multiline-59",
                    "root": "MuiInput-root-53 ChipInput-inputRoot-2",
                    "underline": "MuiInput-underline-57",
                  }
                }
                disableUnderline={true}
                inputRef={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                onKeyUp={[Function]}
                placeholder={null}
                value=""
              >
                <WithStyles(InputBase)
                  classes={
                    Object {
                      "disabled": "MuiInput-disabled-56",
                      "error": "MuiInput-error-58",
                      "focused": "MuiInput-focused-55",
                      "formControl": "MuiInput-formControl-54",
                      "fullWidth": "MuiInput-fullWidth-60",
                      "input": "MuiInput-input-61 ChipInput-input-3",
                      "inputMarginDense": "MuiInput-inputMarginDense-62",
                      "inputMultiline": "MuiInput-inputMultiline-63",
                      "inputType": "MuiInput-inputType-64",
                      "inputTypeSearch": "MuiInput-inputTypeSearch-65",
                      "multiline": "MuiInput-multiline-59",
                      "root": "MuiInput-root-53 ChipInput-inputRoot-2",
                      "underline": null,
                    }
                  }
                  fullWidth={false}
                  inputComponent="input"
                  inputRef={[Function]}
                  multiline={false}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onKeyUp={[Function]}
                  placeholder={null}
                  type="text"
                  value=""
                >
                  <InputBase
                    classes={
                      Object {
                        "adornedEnd": "MuiInputBase-adornedEnd-71",
                        "adornedStart": "MuiInputBase-adornedStart-70",
                        "disabled": "MuiInputBase-disabled-69 MuiInput-disabled-56",
                        "error": "MuiInputBase-error-72 MuiInput-error-58",
                        "focused": "MuiInputBase-focused-68 MuiInput-focused-55",
                        "formControl": "MuiInputBase-formControl-67 MuiInput-formControl-54",
                        "fullWidth": "MuiInputBase-fullWidth-75 MuiInput-fullWidth-60",
                        "input": "MuiInputBase-input-76 MuiInput-input-61 ChipInput-input-3",
                        "inputAdornedEnd": "MuiInputBase-inputAdornedEnd-82",
                        "inputAdornedStart": "MuiInputBase-inputAdornedStart-81",
                        "inputMarginDense": "MuiInputBase-inputMarginDense-77 MuiInput-inputMarginDense-62",
                        "inputMultiline": "MuiInputBase-inputMultiline-78 MuiInput-inputMultiline-63",
                        "inputType": "MuiInputBase-inputType-79 MuiInput-inputType-64",
                        "inputTypeSearch": "MuiInputBase-inputTypeSearch-80 MuiInput-inputTypeSearch-65",
                        "marginDense": "MuiInputBase-marginDense-73",
                        "multiline": "MuiInputBase-multiline-74 MuiInput-multiline-59",
                        "root": "MuiInputBase-root-66 MuiInput-root-53 ChipInput-inputRoot-2",
                      }
                    }
                    fullWidth={false}
                    inputComponent="input"
                    inputRef={[Function]}
                    multiline={false}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyPress={[Function]}
                    onKeyUp={[Function]}
                    placeholder={null}
                    type="text"
                    value=""
                  >
                    <div
                      className="MuiInputBase-root-66 MuiInput-root-53 ChipInput-inputRoot-2 MuiInputBase-formControl-67 MuiInput-formControl-54"
                      onClick={[Function]}
                      onKeyPress={[Function]}
                    >
                      <input
                        aria-invalid={false}
                        className="MuiInputBase-input-76 MuiInput-input-61 ChipInput-input-3"
                        disabled={false}
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        placeholder={null}
                        required={false}
                        type="text"
                        value=""
                      />
                    </div>
                  </InputBase>
                </WithStyles(InputBase)>
              </Input>
            </WithStyles(Input)>
          </div>
        </div>
      </FormControl>
    </WithStyles(FormControl)>
  </ChipInput>
</WithStyles(ChipInput)>
`;
